--- origsrc/xine-lib-1.1.15/configure.ac	2008-08-14 21:51:49.825825800 -0500
+++ src/xine-lib-1.1.15/configure.ac	2008-08-14 21:53:29.903950800 -0500
@@ -2143,7 +2143,7 @@
 
 AC_SYS_LARGEFILE
 AC_CHECK_LIB(posix4, sched_get_priority_min)
-AC_CHECK_FUNCS([vsscanf sigaction sigset getpwuid_r nanosleep lstat memset readlink strchr tzset va_copy])
+AC_CHECK_FUNCS([vsscanf sigaction sigset getpwuid_r nanosleep lstat memset readlink strchr strcasestr tzset va_copy])
 AC_CHECK_FUNCS([snprintf _snprintf], [some_snprintf="yes"; break])
 AC_CHECK_FUNCS([vsnprintf _vsnprintf], [some_vsnprintf="yes"; break])
 AC_CHECK_FUNCS([strcasecmp _stricmp], [some_strcasecmp="yes"; break])
--- origsrc/xine-lib-1.1.15/src/libsputext/demux_sputext.c	2008-07-12 17:23:52.000000000 -0500
+++ src/xine-lib-1.1.15/src/libsputext/demux_sputext.c	2008-08-14 21:55:21.044575800 -0500
@@ -718,6 +718,9 @@
   unsigned a1, a2, a3, a4, b1, b2, b3, b4, comment = 0;
   static unsigned jacoTimeres = 30;
   static int jacoShift = 0;
+#ifndef HAVE_STRCASESTR
+  int cont, jLength;
+#endif
 
   memset(current, 0, sizeof(subtitle_t));
   while (!current->text[0]) {
@@ -805,6 +808,14 @@
     if (isalpha(*p)||*p == '[') {
       if (sscanf(p, "%s %" LINE_LEN_QUOT "[^\n\r]", directive, line1) < 2)
 	return ERR;
+#ifndef HAVE_STRCASESTR
+	jLength = strlen(directive);
+	for (cont = 0; cont < jLength; ++cont) {
+		if (isalpha(*(directive + cont)))
+			*(directive + cont) = toupper(*(directive + cont));
+	}
+#define strcasestr strstr
+#endif
       if ((strcasestr(directive, "RDB") != NULL)
 	  || (strcasestr(directive, "RDC") != NULL)
 	  || (strcasestr(directive, "RLB") != NULL)
