--- origsrc/xine-lib-1.1.15/configure.ac	2008-08-14 23:45:28.615285800 -0500
+++ src/xine-lib-1.1.15/configure.ac	2008-08-15 00:31:54.257334100 -0500
@@ -2680,7 +2680,7 @@
 
 
 dnl Common cflags for all platforms
-CFLAGS="-D_REENTRANT -D_FILE_OFFSET_BITS=64 -DXINE_COMPILE \$(MULTIPASS_CFLAGS) $warnflags $CFLAGS"
+CFLAGS="-D_REENTRANT -D_FILE_OFFSET_BITS=64 -DXINE_COMPILE $warnflags $CFLAGS"
 DEBUG_CFLAGS="-D_REENTRANT -D_FILE_OFFSET_BITS=64 -DXINE_COMPILE $DEBUG_CFLAGS"
 
 if test "x$enable_debug" = "xyes"; then
