--- origsrc/xine-lib-1.2.9/src/input/input_cdda.c	2018-01-11 06:49:47.000000000 -0600
+++ src/xine-lib-1.2.9/src/input/input_cdda.c	2018-04-10 14:36:18.623114400 -0500
@@ -43,10 +43,10 @@
 #  include <alloca.h>
 #endif
 
-#ifdef HAVE_SYS_IOCTL_H
+#if defined(HAVE_SYS_IOCTL_H) && !defined(__CYGWIN__)
 #  include <sys/ioctl.h>
 #endif
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 #  include <windows.h>
 #  include <windef.h>
 #  include <winioctl.h>
@@ -80,7 +80,7 @@
 
 #if defined(__sun)
 #define	DEFAULT_CDDA_DEVICE	"/vol/dev/aliases/cdrom0"
-#elif defined(WIN32)
+#elif defined(WIN32) || defined(__CYGWIN__)
 #define DEFAULT_CDDA_DEVICE "d:\\"
 #elif defined(__OpenBSD__)
 #define	DEFAULT_CDDA_DEVICE	"/dev/rcd0c"
@@ -308,7 +308,7 @@ typedef struct {
   int                  cache_first;
   int                  cache_last;
 
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
   HANDLE  h_device_handle;   /* vcd device descriptor */
   HMODULE hASPI;             /* wnaspi32.dll */
   short i_sid;
@@ -340,7 +340,7 @@ typedef struct {
 } cdda_input_class_t;
 
 
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 
 /* size of a CD sector */
 #define CD_SECTOR_SIZE 2048
@@ -946,7 +946,7 @@ static int read_cdrom_frames(cdda_input_
   return 0;
 }
 
-#elif defined(WIN32)
+#elif defined(WIN32) || defined(__CYGWIN__)
 
 static int read_cdrom_toc(cdda_input_plugin_t *this_gen, cdrom_toc *toc) {
 
@@ -1096,7 +1096,7 @@ static int read_cdrom_frames(cdda_input_
 #define _BUFSIZ 300
 
 
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__CYGWIN__)
 static int parse_url (char *urlbuf, char** host, int *port) {
   char   *start = NULL;
   char   *portcolon = NULL;
@@ -1182,7 +1182,7 @@ network_command( xine_stream_t *stream,
 }
 
 
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__CYGWIN__)
 static int network_connect(xine_stream_t *stream, const char *got_url )
 {
   char *host;
@@ -2036,7 +2036,7 @@ static void _cdda_free_cddb_info(cdda_in
 
 static int cdda_open(cdda_input_plugin_t *this_gen,
 					 const char *cdda_device, cdrom_toc *toc, int *fdd) {
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__CYGWIN__)
   int fd = -1;
 
   if ( !cdda_device ) return -1;
@@ -2240,7 +2240,7 @@ static int cdda_close(cdda_input_plugin_
     close(this_gen->net_fd);
   this_gen->net_fd = -1;
 
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
   if( this_gen->h_device_handle )
      CloseHandle( this_gen->h_device_handle );
   this_gen->h_device_handle = NULL;
@@ -2282,7 +2282,7 @@ static off_t cdda_plugin_read (input_plu
     if( this->cache_last > this->last_frame )
       this->cache_last = this->last_frame;
 
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__CYGWIN__)
     if ( this->fd != -1 )
 #else
 	if ( this->h_device_handle )
@@ -2408,7 +2408,7 @@ static int cdda_plugin_open (input_plugi
   else
     cdda_device = class->cdda_device;
 
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__CYGWIN__)
   if( strchr(cdda_device,':') ) {
     fd = network_connect(this->stream, cdda_device);
     if( fd != -1 ) {
@@ -2426,7 +2426,7 @@ static int cdda_plugin_open (input_plugi
       return 0;
     }
 
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__CYGWIN__)
     err = read_cdrom_toc(this->fd, toc);
 #else
     err = read_cdrom_toc(this, toc);
@@ -2590,7 +2590,7 @@ static const char * const * cdda_class_g
     ip->net_fd = -1;
   }
 
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__CYGWIN__)
   if( strchr(this->cdda_device,':') ) {
     fd = network_connect(ip->stream, this->cdda_device);
     if( fd != -1 ) {
@@ -2608,7 +2608,7 @@ static const char * const * cdda_class_g
       return NULL;
     }
 
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__CYGWIN__)
     err = read_cdrom_toc(fd, toc);
 #else
     err = read_cdrom_toc(ip, toc);
